### PR TITLE
Improvements in install script

### PR DESCRIPTION
**Pull request by zjeffer**
I made some improvements to the install script that made it easier for me to debug problems, so I thought I'd share them.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -24,19 +36,27 @@ HYPRLAND_PATH="$HYPRLOAD_PATH/include/hyprland"
 progress "[3/7] Setting up hyprland source in $HYPRLAND_PATH"
 
 if [ -d "$HYPRLAND_PATH" ]; then
-    git -C "$HYPRLAND_PATH" pull
+	info "Updating existing hyprland source"
+	git -C "$HYPRLAND_PATH" pull origin main
```
Comment (zjeffer) [inline install.sh:40]: I noticed git pull doesn't always seem to work (when already checked out to a specific commit), specifying `origin main` fixed that.

**Comment by leftas** [comment]
While you are at it, when checking out the specific commit, make sure to update the submodules, as well. 
Probably this should suffice: `git -C "$HYPRLAND_PATH" submodule update --recursive`

Current flow - is that if git has updated to the newest wlroots or other submodules, which also has some breaking changes with regard to checked out version, then the script may fail to build hyprland. For example, as of now, we have v0.31 version and current git version which fails to build because wlroots, which are used in git version, made some breaking changes for v0.31 version.

**Comment by zjeffer** [comment]
Also fixed #27 in latest commit.

**Comment by zjeffer** [comment]
Closing due to deprecation

**Pull request closed without merging**